src/app/pages/your-virtrolio/your-virtrolio.component.ts

```diff
@@ -73,6 +93,42 @@ export class YourVirtrolioComponent implements OnInit {
     }
   }
 
+  shareLink() {
+    if (this.canShare()) {
+      this.navigator.share({
+        title: this.displayName + '\'s virtrolio!',
+        text: 'Visit this link to sign their virtrolio and send them a custom message!',
+        url: this.link,
+      })
+        .then(() => console.log('Successful share'))
+        .catch((error) => console.log('Error sharing', error));
+    } else {
+      alert('Share not supported on this device. Use a mobile device!');
+    }
+  }
+
+  /**
+   * Share the user's sharing link on the appropriate platform
+   * @param platform - platform on which to post the sharing link
+   */
+  postOnSocial(platform: string) {
+    const urlFriendlyLink = 'https%3A//virtrolio.web.app/signing?uid=' + this.authService.uid() + '%26key=' +
+      this.link.match(/key=([^&]*)/)[1];
+
+    const bodyText = 'Send%20me%20a%20message!%20';
+
+    if (platform === 'facebook') {
+      window.open('https://www.facebook.com/sharer/sharer.php?u=' + urlFriendlyLink, '_blank');
+    } else if (platform === 'twitter') {
+      window.open('https://twitter.com/intent/tweet?url=' + urlFriendlyLink + '&text=' + bodyText, '_blank');
+    } else if (platform === 'gmail') {
+      window.open('https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&source=mailto&su=' + 'Virtrolio%20-%20Online%20Yearbook%20Signing!'
```

Comment:
Can't you use mailto? This sort of link restricts them to only share with the 1st user account, which isn't ideal.

Reply:
I'll change it to 'email' and use a mailto link